src/H5Dvirtual.c

```diff
@@ -70,6 +71,50 @@
 /* Default size for sub_dset array */
 #define H5D_VIRTUAL_DEF_SUB_DSET_SIZE 128
 
+/*
+ * Determines whether a virtual dataset mapping entry should be inserted
+ * into the R-tree spatial index.
+ *
+ * Parameters:
+ *   entry        - H5O_storage_virtual_ent_t* mapping entry to check
+ *   should_insert - bool output parameter set to true/false
+ *
+ */
+#define H5D_RTREE_SHOULD_INSERT(entry, should_insert)                                                        \
+    do {                                                                                                     \
+        H5S_t  *_vspace      = NULL;                                                                         \
+        H5S_t  *_src_space   = NULL;                                                                         \
+        hsize_t _virt_nelems = 0;                                                                            \
+        hsize_t _src_nelems  = 0;                                                                            \
+                                                                                                             \
+        assert(entry);                                                                                       \
+                                                                                                             \
+        /* Get virtual space and element count */                                                            \
+        if ((_vspace = (entry)->source_dset.virtual_select) != NULL)                                         \
```

Comment:
I believe all of these checks except ndims < 1 can be accomplished with simply `if ((entry)->unlim_dim_virtual >= 0)`